subcommand/control-plane/command.go

```diff
@@ -411,6 +413,7 @@ func (c *Command) constructGatewayProxyRegistration(taskMeta awsutil.ECSTaskMeta
 	gatewaySvc := c.config.Gateway.ToConsulType()
 	gatewaySvc.ID = serviceID
 	gatewaySvc.Service = serviceName
+	gatewaySvc.Address = taskMeta.NodeIP()
```

Comment:
I _think_ this is right. 

On Kube, they had always set the Service.Address to the pod ip for mesh gateways:

* Agentful (0.49.x): https://github.com/hashicorp/consul-k8s/blob/1a17d99134c348dd4f3a4a6a321841589f7ab69b/charts/consul/templates/mesh-gateway-deployment.yaml#L223 - I think this is because it cannot inherit the node address because on Kube they have many proxies on a particular node.
* Agentless (1.2.x): https://github.com/hashicorp/consul-k8s/blob/db1359fc4415c0ef53584f4e75f0d1172dc4e1ba/control-plane/connect-inject/controllers/endpoints/endpoints_controller.go#L671

On ECS, a mesh gateway service currently inherits the node address if unset I think, which was fine because we have a 1:1 mapping of a node/agent to a service instance. Without agents, I think we do need to set this to the task IP address - same as kube is doing.

cc @cthain to double check

Reply:
:+1: Yes this looks correct to me. In a world with client agents (the current ECS implementation) we don't set the `Address` field on the service/proxy registration, which means that it [inherits the agent's address](https://developer.hashicorp.com/consul/api-docs/agent/service#address) (i.e., the task IP). So I think now, for both agentful and agentless, we want to always set the `Address` to the task IP.

Reply:
Thanks for clarifying this.